products/BellHybrid/apps/application-bell-settings/presenter/alarm_settings/PrewakeUpPresenter.cpp

```diff
@@ -30,7 +30,9 @@ namespace app::bell_settings
         this->provider->onVolumeExit   = [this](const auto &) { this->stopSound(); };
         this->provider->onVolumeChange = [this, playSound](const auto &val) {
             this->audioModel->setVolume(val, AbstractAudioModel::PlaybackType::PreWakeup, {});
-            playSound(currentSoundPath);
+            if (this->audioModel->hasPlaybackFinished()) {
```

Comment:
why only the prewakeup presenter checks if the playback has finished?

Reply:
fixed